lib/serializer.ts

```diff
@@ -5,11 +5,12 @@ import * as utils from "./util/utils";
 import { duration, isDuration } from "moment";
 const isBuffer = require("is-buffer");
 import * as isStream from "is-stream";
+import { isArray } from "util";
```

Comment:
We don't need this. Be vary of importing the util library from node.js This has a direct impact on the bundle size in the browser. There is a function on the Array type `Array.isArray()`, use that instead.

Reply:
Looking into removing all the "util" usages--would really like a promisify alternative though.

Reply:
Looks like it's simply `new Promise(function(resolve, reject) { ...})` :)

Reply:
We have never used node.js's util library at all. 
Note: We have a util folder in the runtime and importing stuff from there is fine. Just making sure there is no confusion.
And why would we ever need a Promisified version of `Array.isArray()` ?

Reply:
`Array.isArray()` is a synchronous call.

Reply:
I was making use of `promisify` from within the util module in another file :)

Reply:
Ahh. I see. Yup that is super simple.
```
return new Promise((resolve, reject) => {
  if (err) reject(err);

  resolve(result);
});
```

That is what we are essentially doing in the generated code when we say we support callback and promises both [for example](https://github.com/Azure/azure-sdk-for-node/blob/master/lib/services/storageManagement2/lib/operations/storageAccounts.js#L2994-L3000).

Reply:
It is better to use arrow functions `() => {}` over `function () {}` since the `this` is preserved inside the arrow function

```javascript
function foo() {
  this.name = "foo";
   const bar = () => {
      console.log(this.name); // foo
   };
}
```

Reply:
v/s 
```javascript
function foo() {
  this.name = "foo";
  let self = this;
   function bar() => {
      console.log(this.name); // undefined
      console.log(self.name); //"foo"
   };
}
```